### PR TITLE
Create zkwe.csl

### PR DESCRIPTION
**Pull request by 540felix**
Citation style for ZKwE (http://www.iee.put.poznan.pl/zkwe/)

### CODE REVIEW

**Comment by rmzelle** [comment]
Our Travis tests fail because your file name is missing the required ".csl" extension.

Can you visit https://github.com/540felix/styles/blob/patch-3/zkwe, click the pencil icon, and change the file name from "zkwe" to "zastosowania-komputerow-w-elektrotechnice.csl"? Thanks.

**Comment by rmzelle** [comment]
See also See also https://help.github.com/articles/renaming-a-file/

**Comment by 540felix** [comment]
Thank you for your advice, it was very helpful.

File: zastosowania-komputerow-w-elektrotechnice.csl [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+      <name>Michael Berkowitz</name>
+      <email>mberkowi@gmu.edu</email>
+    </contributor>
+    <category citation-format="numeric"/>
+    <category field="engineering"/>
+    <category field="generic-base"/>
+    <updated>2015-04-13T10:05:52+00:00</updated>
+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
+  </info>
+  <locale xml:lang="pl-PL">
```
Comment (adam3smith) [inline zastosowania-komputerow-w-elektrotechnice.csl]: this seems a bit odd: you're defining a couple of terms for Polish, where everything else (things like editor, e.g.) will be in English. What's the expectation for this style? Are papers going to be in Polish, English, or both and if both, does the citation style differ?

File: zastosowania-komputerow-w-elektrotechnice.csl [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+      <name>Mateusz Pacek</name>
+      <email>mateusz.m.pacek@student.put.poznan.pl</email>
+    </author>
+    <contributor>
+      <name>Michael Berkowitz</name>
+      <email>mberkowi@gmu.edu</email>
+    </contributor>
+    <category citation-format="numeric"/>
+    <category field="engineering"/>
+    <category field="generic-base"/>
```
Comment (adam3smith) [inline zastosowania-komputerow-w-elektrotechnice.csl]: could you remove this line -- the style isn't "generic" -- it's an engineering style.

File: zastosowania-komputerow-w-elektrotechnice.csl [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+      </if>
+      <else>
+        <text variable="title" quotes="false"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <choose>
+      <if type=" bill book chapter graphic legal_case legislation motion_picture paper-conference song" match="any">
+        <group delimiter=": ">
```
Comment (adam3smith) [inline zastosowania-komputerow-w-elektrotechnice.csl]: a) is this really what you want? just the place of publication for the these item types, just the publisher for the others?
b) even if, could you remove both this group and the one in 121/123 below? Groups with only a single element have no effect (and the delimiter doesn't do anything, obviously).

**Comment by adam3smith** [comment]
Please see my questions inline.

File: zastosowania-komputerow-w-elektrotechnice.csl (hunk 1 above)
Comment (540felix) [inline zastosowania-komputerow-w-elektrotechnice.csl]: As far as I know, papers are going to be both Polish and English, and the citation style doesn't differ.

File: zastosowania-komputerow-w-elektrotechnice.csl (hunk 2 above)
Comment (540felix) [inline zastosowania-komputerow-w-elektrotechnice.csl]: Done.

File: zastosowania-komputerow-w-elektrotechnice.csl (hunk 3 above)
Comment (540felix) [inline zastosowania-komputerow-w-elektrotechnice.csl]: Ad. a) I've changed the second one to publisher-place.
Ad. b) Removed.
Thanks for your help!

File: zastosowania-komputerow-w-elektrotechnice.csl [hunk 4]
```diff
@@ -0,0 +1,321 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="sort-only">
+  <info>
+    <title>Zastosowania Komputerów w Elektrotechnice</title>
+    <title-short>ZKwE</title-short>
+    <id>http://www.zotero.org/styles/zastosowania-komputerow-w-elektrotechnice</id>
+    <link href="http://www.zotero.org/styles/zastosowania-komputerow-w-elektrotechnice" rel="self"/>
+    <link href="http://www.zotero.org/styles/ieee" rel="template"/>
+    <link href="http://www.iee.put.poznan.pl/caiee/redakcja.doc" rel="documentation"/>
```
Comment (adam3smith) [inline zastosowania-komputerow-w-elektrotechnice.csl]: that link should probably be http://www.iee.put.poznan.pl/zkwe/download/guide.doc
this one goes to a 404 for me.

File: zastosowania-komputerow-w-elektrotechnice.csl [hunk 5]
```diff
@@ -0,0 +1,321 @@
+<?xml version="1.0" encoding="utf-8"?>
+<style xmlns="http://purl.org/net/xbiblio/csl" class="in-text" version="1.0" demote-non-dropping-particle="sort-only">
```
Comment (adam3smith) [inline zastosowania-komputerow-w-elektrotechnice.csl]: So, given that you have hardcoded some terms into English into the style, I suggest we set the style to English as a default: add ```default-locale="en-US"``` to this line (line 2).

File: zastosowania-komputerow-w-elektrotechnice.csl [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+    <contributor>
+      <name>Michael Berkowitz</name>
+      <email>mberkowi@gmu.edu</email>
+    </contributor>
+    <category citation-format="numeric"/>
+    <category field="engineering"/>
+    <updated>2015-04-13T10:05:52+00:00</updated>
+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
+  </info>
+  <locale xml:lang="pl-PL">
```
Comment (adam3smith) [inline zastosowania-komputerow-w-elektrotechnice.csl]: and then change the language here to "en-US" so these changes to terms will actually get picked up.

Comment (540felix) [inline zastosowania-komputerow-w-elektrotechnice.csl]: All right, everything's done. Thanks a lot for your time and contribution in my project!

**Comment by adam3smith** [comment]
Great, thanks!

**Pull request merged**